agr_literature_service/lit_processing/dump_json_data.py

```diff
@@ -0,0 +1,576 @@
+
+    try:
+        jsonData = {"data": data,
+                    "metaData": metaData}
+        fw = open(filename_with_path, 'w')
+        fw.write(json.dumps(jsonData, indent=4, sort_keys=True))
+        fw.close()
+    except Exception as e:
+        print("Error when generating", filename_with_path, ": " + str(e))
+        exit()
+
+
+def upload_json_file_to_s3(json_path, json_file, datestamp, ondemand):
+
+    env_state = environ.get('ENV_STATE', 'develop')
+    if env_state == 'build':
+        env_state = 'develop'
+    if env_state == 'test':
+        return
+
+    json_file_with_datestamp = json_file.replace('.json', '_' + datestamp + '.json')
+
+    if ondemand:
+        s3_filename = ondemand_bucket.replace('develop', env_state) + json_file_with_datestamp
+        upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+        return
+
+    ## upload file to recent bucket
+    s3_filename = recent_bucket.replace('develop', env_state) + json_file_with_datestamp
+    upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+
+    ## upload file to latest bucket
+    s3_filename = latest_bucket.replace('develop', env_state) + json_file
+    upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename)
+
+    ## upload file to monthly bucket if it is first day of the month
+    todayDate = date.today()
+    if todayDate.day == 1:
+        s3_filename = monthly_bucket.replace('develop', env_state) + json_file_with_datestamp
+        upload_file_to_s3(json_path + json_file, s3_bucket, s3_filename, 'GLACIER_IR')
+
+
+def concatenate_json_files(json_file, index):
+
+    if index == 1:
+        rename(json_file + '_0', json_file)
+        return
+
+    fw = open(json_file, "w")
+
+    for i in range(index):
+        this_json_file = json_file + '_' + str(i)
+        f = open(this_json_file)
+        if i == 0:
+            # first chuck of data so keep beginning part, remove ending part
+            for line in f:
+                if line.strip().endswith("}") and len(line) == 10:
+                    fw.write("        },\n")
+                    break
+                fw.write(line)
+        elif i + 1 == index:
+            # last chunk of data so keep ending part, remove beginning part
+            for line in f:
+                if line.startswith('{') or line.startswith('    "data": ['):
+                    continue
+                fw.write(line)
+        else:
+            # mid section(s) so remove beginning part & ending part
+            for line in f:
+                if line.strip().endswith("}") and len(line) == 10:
+                    fw.write("        },\n")
+                    break
+                if line.startswith('{') or line.startswith('    "data": ['):
+                    continue
+                fw.write(line)
+        f.close()
+
+        remove(this_json_file)
+
+    fw.close()
+
+
+def get_meta_data(mod, datestamp):
+
+    ## return more info here?
```

Comment:
I think it's fine, I can't think of what else we might want, until we started creating database model versions or something like that.

Reply:
OK sounds good!